showcase/src/app/shared/effects/stark-error-handler.effects.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Injector } from "@angular/core";
+import { Actions, Effect, ofType } from "@ngrx/effects";
+import { Observable } from "rxjs";
+import { map, take } from "rxjs/operators";
+
+import { STARK_TOAST_NOTIFICATION_SERVICE, StarkMessageType, StarkToastNotificationService } from "@nationalbankbelgium/stark-ui";
+import { StarkErrorHandlerActionTypes, StarkUnhandledError } from "@nationalbankbelgium/stark-core";
+
+/**
+ * Unique Id of the displayed toaster
+ */
+const _uniqueId: Function = require("lodash/uniqueId");
+
+/**
+ * This class is used to determine what to do with an error
+ */
+@Injectable()
+export class StarkErrorHandlerEffects {
+	private _starkToastNotificationService: StarkToastNotificationService;
+
+	/**
+	 * Class constructor
+	 * @param actions$ - the action to perform
+	 * @param injector - the injector of the class
+	 */
+	public constructor(private actions$: Actions, private injector: Injector) {}
+
+	@Effect({ dispatch: false })
+	public setToastMessage$(): Observable<void> {
+		return this.actions$.pipe(
+			ofType<StarkUnhandledError>(StarkErrorHandlerActionTypes.UNHANDLED_ERROR),
+			map((action: StarkUnhandledError) => {
+				this.toastNotificationService
```

Comment:
As we discussed, the issue about the toast not being shown until some UI change occurs is related to the fact that the [everything executed inside the ErrorHandler does not invoke a change detection cycle](https://github.com/angular/material2/issues/7233).

So, another guy proposed a [solution for this](https://github.com/angular/material2/issues/13181). I've tested it and it works as expected 😉